### PR TITLE
Hw13 10 2020 from Veronika Silina

### PR DESCRIPTION
**Pull request by veronika-silina**
Homework from the lesson 13-10-2020 with tests

https://github.com/tel-ran-de/BE12-Basics/issues/6

### CODE REVIEW

File: src/lessonsExamples/hw1310/StringTransform.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+public class StringTransform {
+    ////"abc defG ikl" -> "ABC defG IKL"
+    //    // length == 3 -> toUpperCase
+    //   // length == 5 -> toLowerCase
+    //   // length is even -> to UpperCase
+
+    static String transform(String sentence, Predicate<Integer> predicate, UnaryOperator<String> function) {
```
Comment (sergeylukichev) [inline src/lessonsExamples/hw1310/StringTransform.java:12]: You may write a unit test for this method only with different lambda expressions.

File: src/lessonsExamples/lesson0610/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package lessonsExamples.lesson0610;
+
+import java.util.*;
+
+public class Anagram {
+    private static List<String> VOC = Arrays.asList("anna", "ivan", "naan", "vani", "piotr", "navi");
+
+    public static List<String> getAnagram(String word) {
+        List<String> res = new ArrayList<>();
+        for (String w : VOC) {
```
Comment (sergeylukichev) [inline src/lessonsExamples/lesson0610/Anagram.java:10]: This is a good solution, but we considered a better (faster) one which uses maps.

File: src/lessonsExamples/lesson0610/Anagram.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            }
+        }
+        return res;
+    }
+
+    public static Map<String, List<String>> buildMap(List<String> voc){
+        Map<String, List<String>> result = new HashMap<>();
+        for (String w : voc) {
+            if (result.get(sortWord(w)) == null) {
+
```
Comment (sergeylukichev) [inline src/lessonsExamples/lesson0610/Anagram.java:22]: seems incomplete.

**Review by sergeylukichev — APPROVED** [review]
👍

File: src/lessonsExamples/lesson0610/Anagram.java (hunk 3 above)
Comment (veronika-silina) [reply src/lessonsExamples/lesson0610/Anagram.java:22]: Sorry, it looks like this file shouldn't be here, I'll try to remove it from this commit

**Pull request merged**